SeleniumDemoQA/NUnitHomeworks/ArrayListHomework.cs

```diff
@@ -0,0 +1,167 @@
+            }
+            // Assert
+            Assert.That(reversedArray, Is.EqualTo(expectedReversed));
+        }
+
+        [Test]
+        public void TestReplaceStudentName()
+        {
+            // Arrange
+            string[] students = new string[] { "John", "Jane", "Bill", "Sue", "Anna" };
+            string newName = "Bob";
+
+
+            students[1] = newName;
+
+            // Act
+            // Implement replacement
+
+            // Assert
+            Assert.That(students[1], Is.EqualTo(newName)); // Second student's name should be "Bob"
+        }
+
+        [TestCase("Sue", true)]
+        [TestCase("Alex", false)]
+        public void TestStudentNameExists(string nameToCheck, bool existsExpected)
+        {
+            // Arrange
+            string[] students = new string[] { "John", "Jane", "Bill", "Sue", "Anna" };
+
+            // Act
+            bool exists = false; // Replace false with code to get to know if nameToCheck exists in students []
+            
+
+            foreach (string student in students)
+            {
+                if (student.Equals(nameToCheck)) {
+                    exists = true; 
+                    break;
+                }   
+            }
+
+            // Assert
+            Assert.That(exists, Is.EqualTo(existsExpected));
+        }
+
+        [Test]
+        public void TestDoubleValues()
+        {
+            // Arrange
+            List<int> numbers = new List<int> { 1, 2, 3, 4, 5 };
+            List<int> expectedDoubled = new List<int> { 2, 4, 6, 8, 10 };
+
+            // Act
+            List<int> doubledNumbers = new List <int> ();
+
+            foreach ( int number in numbers)
+            {
+                doubledNumbers.Add(number*2);
+            }
+
+
+            // Implement code to fill doubledNumbers with values
+
+            // Assert
+            Assert.That(doubledNumbers, Is.EqualTo(expectedDoubled));
+        }
+
+        [Test]
+        public void TestRemoveEvenNumbers()
+        {
+            // Arrange
+            List<int> numbers = new List<int> { 1, 2, 3, 4, 5 };
+            List<int> expectedList = new List<int> { 1, 3, 5 };
+
+            // Act
+            // Implement code to remove extra items
+           for (int i = 0; i < numbers.Count;)
+           {
+                if (numbers[i] % 2 == 0) {
+                    numbers.Remove(numbers[i]);
+                } else
+                {
+                    i++;
+                }
+           }
+           
+
+            // Assert
+            Assert.That(numbers, Is.EqualTo(expectedList));
+        }
+
+
+        [Test]
+        public void TestAddAnimal()
+        {
+            // Arrange
+            List<string> animals = new List<string> { "Lion", "Tiger", "Elephant", "Giraffe", "Zebra" };
+            string newAnimal = "Penguin";
+            List<string> expectedList = new List<string> { "Lion", "Tiger", "Elephant", "Giraffe", "Zebra", "Penguin" };
```

Comment:
it is not expected that you modify expectedList. You need to write code below in order to insert newAnimal into correct place in the List, but not to move "Penguin" from the middle to the end in order to just use Add()